x/blob/ante/blob_share_decorator.go

```diff
@@ -25,12 +26,17 @@ func NewBlobShareDecorator(k BlobKeeper) BlobShareDecorator {
 // returns an error if tx contains a MsgPayForBlobs where the shares occupied by
 // the PFB exceeds the max number of shares in a data square.
 func (d BlobShareDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
-	if !ctx.IsCheckTx() {
-		return next(ctx, tx, simulate)
-	}
-
-	if ctx.BlockHeader().Version.App == v1.Version {
+	switch ctx.BlockHeader().Version.App {
+	case v2.Version:
+		if !ctx.IsCheckTx() {
+			return next(ctx, tx, simulate)
+		}
+	case v1.Version:
 		return next(ctx, tx, simulate)
+	default:
+		if ctx.IsReCheckTx() {
+			return next(ctx, tx, simulate)
+		}
```

Comment:
[question] this appears to be missing a `case v3.Version`

[nit] it's cognitively difficult to understand if this case block is correct. Is it really necessary to skip the ante handler for recheck tx? That seems like a minor performance enhancement and/or premature optimization. 

I would expect the switch statement to look like this:

In other words, not depend on ReCheckTx or CheckTx
```go
	switch ctx.BlockHeader().Version.App {
	case v1.Version:
		// this ante handler was not used in v1
		return next(ctx, tx, simulate)
	case v2.Version:
		// run ante handler
	case v3.Version:
		// run ante handler
	default:
		return ctx, fmt.Errorf("unsupported app version %v", ctx.BlockHeader().Version.App)
	}
```